cvmfs/swissknife_lease_curl.h

```diff
@@ -0,0 +1,29 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef CVMFS_SWISSKNIFE_LEASE_CURL_H_
+#define CVMFS_SWISSKNIFE_LEASE_CURL_H_
+
+#include <string>
+
+#include "curl/curl.h"
+
+struct CurlBuffer {
+  std::string data;
+};
+
+size_t RecvCB(void *buffer, size_t size, size_t nmemb, void *userp);
+
+CURL* PrepareCurl(const char* method);
+
+bool MakeAcquireRequest(const std::string& user_name,
```

Comment:
We might want to have a class around it.  Perhaps it can be a single class with the functions of this compilation unit plus the JSON parsing.